caravel/assets/javascripts/SqlLab/components/SouthPane.jsx

```diff
@@ -1,25 +1,42 @@
 import { Alert, Tab, Tabs } from 'react-bootstrap';
 import QueryHistory from './QueryHistory';
 import ResultSet from './ResultSet';
-import { areArraysShallowEqual } from '../../reduxUtils';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as Actions from '../actions';
 import React from 'react';
 
 import shortid from 'shortid';
 
+// editorQueries are queries executed by users
+// passed from SqlEditor component
+// queries are all queries executed
+// passed from global state for the purpose of data preview tabs
 const propTypes = {
-  queries: React.PropTypes.array.isRequired,
+  editorQueries: React.PropTypes.array.isRequired,
+  queries: React.PropTypes.object.isRequired,
   actions: React.PropTypes.object.isRequired,
+  dataPreviewQueryIds: React.PropTypes.array,
+  activeSouthPaneTab: React.PropTypes.string,
+};
+
+const defaultProps = {
+  dataPreviewQueryIds: [],
+  activeSouthPaneTab: 'Results',
 };
 
 class SouthPane extends React.PureComponent {
-  shouldComponentUpdate(nextProps) {
```

Comment:
I think the optimization here is fairly important, and more so as we now have more data in here. 

Here's an alternative that goes in the direction of reducing the number of components that interact with the redux store: in TabbedQueryEditors, prepare a `dataPreviewQueries` array, and pass it to `SqlEditor`, which passes it to `SouthPane`. In SouthPane you can then look for `areArraysShallowEqual` of both `queries` and `dataPreviewQueries`.

This is clearly denormalization-type workload, and I feel like it belongs in the controller (TabbedQueryEditors in this case) and that we should limit the number of components that play that role (interacting with the redux store) as much as possible.

@ascott what do you think?